tests/unittests/unittests.cpp

```diff
@@ -4264,6 +4269,137 @@ class ConcurrentQueueTests : public TestClass<ConcurrentQueueTests>
 		
 		return true;
 	}
+
+	bool acquire_and_signal()
+	{
+		typedef ::moodycamel::LightweightSemaphore semaphore;
+
+		// Test resource acquisition from one other thread.
+		{
+			semaphore s;
+			s.signal(); // Single resource available
+						
+			const auto fnTestSingleAcquire = 
+				[](::moodycamel::LightweightSemaphore &s) -> bool
+				{
+					for (std::size_t k = 0; k < 100; ++k)
+					{
+						ASSERT_OR_FAIL(s.wait(100 * 1000)); // Wait 100 ms
```

Comment:
`ASSERT_OR_FAIL` is really just a fancy macro that does `return false` on failure. (My test framework is pretty basic.) This won't work from within a thread callback, only the test body itself. Instead, an atomic boolean should be set and assertion on after the threads join. (The return value from the `std::thread` entry function cannot be recuperated.)